### PR TITLE
Project Has Been Refactored with New Infrastructure

### PR DESCRIPTION
**Pull request by agitrubard**
This pull request includes code changes to refactor the project with a new infrastructure. The changes were made to make the project run more efficiently and reliably.

**Changes Made**
- Cleaned up the old code structure and replaced it with the new infrastructure.
- Made the code compatible with the new infrastructure.
- Removed unnecessary files.
- Removed unnecessary code.
- Rewrote some of the code.
- Refactored the token infrastructure.
- Refactored authentication/authorization flows.
- Added controller/unit tests.
- Refactored all user flows.

---

# New Infrastructure
ays-be-layer drawio
ays-be-package drawio

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,12 @@
-FROM azul/zulu-openjdk:17
-VOLUME /tmp
-COPY target/*.jar app.jar
-ENTRYPOINT ["java","-jar","/app.jar"]
\ No newline at end of file
+FROM openjdk:17-jdk-slim AS build
```
Comment (skayikci) [inline Dockerfile:1]: for m1 chipped apple laptops, this image will be problem. please use azul/zulu.

File: src/main/java/com/ays/AysApplicationEventListener.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * A listener that logs a message when the AYS application is ready to start.
+ */
+@Slf4j
+@Configuration
+class AysApplicationEventListener implements ApplicationListener<ApplicationReadyEvent> {
```
Comment (skayikci) [inline src/main/java/com/ays/AysApplicationEventListener.java:13]: may i ask why you implemented this just for the sake of logging that the application is up and running?

File: src/main/java/com/ays/admin_user/model/dto/request/AdminUserRegisterRequest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+import com.ays.common.model.AysPhoneNumber;
+import jakarta.validation.constraints.Email;
+import jakarta.validation.constraints.NotBlank;
+import lombok.Builder;
+import lombok.Data;
+
+@Data
+@Builder
+public class AdminUserRegisterRequest {
```
Comment (skayikci) [inline src/main/java/com/ays/admin_user/model/dto/request/AdminUserRegisterRequest.java:11]: missing javadoc

File: src/main/java/com/ays/admin_user/service/impl/AdminUserAuthServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+        if (adminUserRepository.findByEmail(registerRequest.getEmail()).isPresent()) {
+            throw new AysAdminUserAlreadyExistsByEmailException(registerRequest.getEmail());
+        }
+
+        if (adminUserRepository.findByUsername(registerRequest.getUsername()).isPresent()) {
+            throw new AysAdminUserAlreadyExistsByUsernameException(registerRequest.getUsername());
+        }
+
+//        TODO : check phone number!
```
Comment (skayikci) [inline src/main/java/com/ays/admin_user/service/impl/AdminUserAuthServiceImpl.java:78]: please fix the todo if possible, and remove this line.

File: src/main/java/com/ays/user/model/dto/request/UserSaveRequest.java [hunk 5]
```diff
@@ -0,0 +1,6 @@
+package com.ays.user.model.dto.request;
+
+public class UserSaveRequest {
```
Comment (skayikci) [inline src/main/java/com/ays/user/model/dto/request/UserSaveRequest.java:3]: why is the body empty?

File: src/main/java/com/ays/user/model/dto/response/UsersResponse.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import com.ays.user.model.enums.UserRole;
+import com.ays.user.model.enums.UserStatus;
+import lombok.Builder;
+import lombok.Getter;
+
+@Getter
+@Builder
+public class UsersResponse {
+
+    private String username;
```
Comment (skayikci) [inline src/main/java/com/ays/user/model/dto/response/UsersResponse.java:13]:
```suggestion
    private String userName;
```

File: src/main/java/com/ays/user/model/entity/UserEntity.java [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+import lombok.EqualsAndHashCode;
+import lombok.NoArgsConstructor;
+import lombok.experimental.SuperBuilder;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Users entity, which holds the information regarding the system user.
```
Comment (skayikci) [inline src/main/java/com/ays/user/model/entity/UserEntity.java:21]:
```suggestion
 * User entity, which holds the information regarding the system user.
```

File: src/main/java/com/ays/user/service/impl/UserServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+class UserServiceImpl implements UserService {
+
+    private final UserRepository userRepository;
+
+    private final UserEntityToUserMapper userEntityToUserMapper = UserEntityToUserMapper.initialize();
+
+
+    @Transactional
+    public void saveUser(UserSaveRequest saveRequest) {
+//
```
Comment (skayikci) [inline src/main/java/com/ays/user/service/impl/UserServiceImpl.java:32]: Please remove if not used.

File: src/main/java/com/ays/user/service/impl/UserServiceImpl.java [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+    @Override
+    @Transactional
+    public void updateUser(UserUpdateRequest updateRequest) {
+
+        final String id = updateRequest.getId();
+        final UserEntity userEntity = userRepository.findById(id)
+                .orElseThrow(() -> new AysUserNotExistByIdException(id));
+
+        // userEntity.update(); TODO : user update method must be written
```
Comment (skayikci) [inline src/main/java/com/ays/user/service/impl/UserServiceImpl.java:101]: please implement or remove.

File: src/test/java/old/test/user/controller/UserControllerTest.java
Comment (skayikci) [inline src/test/java/old/test/user/controller/UserControllerTest.java:1]: please remove

File: src/test/java/old/test/user/model/UserBuilder.java
Comment (skayikci) [inline src/test/java/old/test/user/model/UserBuilder.java:1]: please remove

File: src/test/java/old/test/user/model/entity/UserEntityBuilder.java
Comment (skayikci) [inline src/test/java/old/test/user/model/entity/UserEntityBuilder.java:1]: please remove

File: src/test/java/old/test/user/service/AuthServiceTest.java
Comment (skayikci) [inline src/test/java/old/test/user/service/AuthServiceTest.java:1]: please remove

File: src/test/java/old/test/user/service/impl/UserServiceImplTest.java
Comment (skayikci) [inline src/test/java/old/test/user/service/impl/UserServiceImplTest.java:1]: please remove

File: Dockerfile (hunk 1 above)
Comment (agitrubard) [reply Dockerfile:1]: Is there a need to run it on Docker locally, or is the Dockerfile only used by the server side?

File: src/main/java/com/ays/AysApplicationEventListener.java (hunk 2 above)
Comment (agitrubard) [reply src/main/java/com/ays/AysApplicationEventListener.java:13]: I have written this class to add a log indicating that the application has been started in addition to the existing logs.

File: src/main/java/com/ays/admin_user/model/dto/request/AdminUserRegisterRequest.java (hunk 3 above)
Comment (agitrubard) [reply src/main/java/com/ays/admin_user/model/dto/request/AdminUserRegisterRequest.java:17]: Fixed ✅

File: src/main/java/com/ays/admin_user/service/impl/AdminUserAuthServiceImpl.java (hunk 4 above)
Comment (agitrubard) [reply src/main/java/com/ays/admin_user/service/impl/AdminUserAuthServiceImpl.java]: Fixed ✅

File: src/main/java/com/ays/user/model/dto/response/UsersResponse.java [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+import com.ays.organization.model.entity.OrganizationEntity;
+import com.ays.user.model.enums.UserRole;
+import com.ays.user.model.enums.UserStatus;
+import lombok.Builder;
+import lombok.Getter;
+
+@Getter
+@Builder
+public class UsersResponse {
```
[the remark this answers is not in the document]
Comment (agitrubard) [reply src/main/java/com/ays/user/model/dto/response/UsersResponse.java:14]: I created this object to return the necessary data in the API that will be used in the user listing screen. That's why I named it `UsersResponse`. On the other hand, the `UserResponse` object is completely different and is used to display detailed information about a user.

File: src/main/java/com/ays/user/model/dto/response/UsersResponse.java (hunk 6 above)
Comment (agitrubard) [reply src/main/java/com/ays/user/model/dto/response/UsersResponse.java:17]: Fixed ✅

File: src/main/java/com/ays/user/service/impl/UserServiceImpl.java (hunk 8 above)
Comment (agitrubard) [reply src/main/java/com/ays/user/service/impl/UserServiceImpl.java]: Fixed ✅

File: src/main/java/com/ays/user/service/impl/UserServiceImpl.java (hunk 9 above)
Comment (agitrubard) [reply src/main/java/com/ays/user/service/impl/UserServiceImpl.java]: Fixed ✅

File: src/test/java/old/test/user/controller/UserControllerTest.java
Comment (agitrubard) [reply src/test/java/old/test/user/controller/UserControllerTest.java:1]: Fixed ✅

File: src/test/java/old/test/user/exception/handler/UserExceptionHandlerTest.java
[the remark this answers is not in the document]
Comment (agitrubard) [reply src/test/java/old/test/user/exception/handler/UserExceptionHandlerTest.java:1]: Fixed ✅

File: src/test/java/old/test/user/service/impl/UserServiceImplTest.java
Comment (agitrubard) [reply src/test/java/old/test/user/service/impl/UserServiceImplTest.java:1]: Fixed ✅

File: src/test/java/old/test/user/service/AuthServiceTest.java
Comment (agitrubard) [reply src/test/java/old/test/user/service/AuthServiceTest.java:1]: Fixed ✅

File: src/test/java/old/test/user/model/entity/UserEntityBuilder.java
Comment (agitrubard) [reply src/test/java/old/test/user/model/entity/UserEntityBuilder.java:1]: Fixed ✅

File: src/test/java/old/test/user/model/UserBuilder.java
Comment (agitrubard) [reply src/test/java/old/test/user/model/UserBuilder.java:1]: Fixed ✅

File: src/test/java/old/test/user/handler/RoleExceptionHandlerTest.java
[the remark this answers is not in the document]
Comment (agitrubard) [reply src/test/java/old/test/user/handler/RoleExceptionHandlerTest.java:1]: Fixed ✅

**Pull request merged**